rasa/core/channels/twilio_voice.py

```diff
@@ -0,0 +1,342 @@
+                    )
+                )
+
+                twilio_response = self.build_twilio_voice_response(collector.messages)
+            # If the user doesn't respond resend the last message.
+            else:
+                # Get last user utterance from tracker.
+                tracker = request.app.agent.tracker_store.retrieve(sender_id)
+                last_response = next(
+                    (e for e in reversed(tracker.events) if isinstance(e, BotUttered)),
+                    None,
+                )
+
+                # If no previous utterance found use the reprompt_fallback phrase.
+                if last_response is None:
+                    last_response = self.reprompt_fallback_phrase
+                else:
+                    last_response = last_response.text
+
+                twilio_response = self.build_twilio_voice_response(
+                    [{"text": last_response}]
+                )
+            return response.text(str(twilio_response), content_type="text/xml")
+
+        return twilio_voice_webhook
+
+    def build_twilio_voice_response(
+        self, messages: List[Dict[Text, Any]]
+    ) -> VoiceResponse:
+        """Builds the Twilio Voice Response object."""
+        voice_response = VoiceResponse()
+        gather = Gather(
+            input="speech",
+            action=f"/webhooks/{self.name()}/webhook",
+            actionOnEmptyResult=True,
+            speechTimeout=self.speech_timeout,
+            speechModel=self.speech_model,
+            enhanced=self.enhanced,
+        )
+
+        # Add pauses between messages.
+        # Add a listener to the last message to listen for user response.
+        for i, message in enumerate(messages):
+            if i + 1 == len(messages):
+                gather.say(message["text"], voice=self.assistant_voice)
+                voice_response.append(gather)
+            else:
+                voice_response.say(message["text"], voice=self.assistant_voice)
+                voice_response.pause(length=1)
+
+        return voice_response
+
+
+class TwilioVoiceCollectingOutputChannel(CollectingOutputChannel):
+    """Output channel that collects send messages in a list.
+
+    (doesn't send them anywhere, just collects them).
+    """
+
+    EMOJI = re.compile(
+                "["
+                "\U0001F600-\U0001F64F"  # emoticons
+                "\U0001F300-\U0001F5FF"  # symbols & pictographs
+                "\U0001F680-\U0001F6FF"  # transport & map symbols
+                "\U0001F1E0-\U0001F1FF"  # flags (iOS)
+                "\U00002702-\U000027B0"
+                "\U000024C2-\U0001F251"
+                "\u200d"  # zero width joiner
+                "\u200c"  # zero width non-joiner
+                "]+",
+                flags=re.UNICODE,
+            )
+
+    @classmethod
+    def name(cls) -> Text:
+        """Name of the output channel."""
+        return "twilio_voice"
+
+    async def emoji_warning(
```

Comment:
no need to make this `async`
```suggestion
    def emoji_warning(
```